policykit/policyengine/engine.py

```diff
@@ -0,0 +1,309 @@
+import logging
+
+from actstream import action as actstream_action
+from django.conf import settings
+
+from policyengine.utils import ActionKind
+
+logger = logging.getLogger(__name__)
+db_logger = logging.getLogger("db")
+
+
+class EvaluationLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        kwargs["extra"] = self.extra
+        return (msg, kwargs)
+
+
+class EvaluationContext:
+    """
+    Class to hold all variables available in a policy evaluation.
+    All attributes on this class are in scope and can be used by the policy author.
+
+    Attributes:
+        proposal (Proposal): The proposal representing this evaluation.
+        action (BaseAction): The action that triggered this policy evaluation.
+        policy (Policy): The policy being evaluated.
+        slack (SlackCommunity)
+        discord (DiscordCommunity)
+        discourse (DiscourseCommunity)
+        reddit (RedditCommunity)
+        metagov (Metagov): Metagov library for performing enabled actions and processes.
+        logger (logging.Logger): Logger that will log messages to the PolicyKit web interface.
+
+    """
```

Comment:
This class isn't actually exposed to the policy author, but we use it to document all the in-scope variables. Any attributes on this class will be available to the policy author.